### ISSUE TITLE
Data d3Hierarchy - value is always taken as sum of children values

### ISSUE THREAD

**Issue by Vity01** [vasturiano/sunburst-chart#14]
Hello,
first of all - thanks for the great graph component and sharing it. 

For our usecase all our nodes *have value attribute set*, but the component always counts the values as a sum of all children. 
```
...sum(accessorFn(state.size));
```
I would like to skip this step and to use our value directly.

**Comment by vasturiano** [comment]
@Vity01 a non-leaf node in the tree needs to take into account the sum of the children's values because this is a hierarchical layout, i.e. a node's radial angle needs to be at least as wide as the sum of its children combined.

What you can do is specify how the value of each node (leaf or non-leaf) is computed, using the `size` property, like:
```
myChart.size(d => d.value * 10)
```

**Comment by Vity01** [comment]
@Vasturiano I am not sure if we understand each other. I don't understand how the settings can bypass the summing of the nodes values. 
The default `size` value is `d => d.value`.

**Comment by Vity01** [comment]
I am referring both :-). We have pre-computed all values for all nodes so the summing is not necessary from my point of view. I think it can be skipped from what I see in the code : 
```
 const hierData = d3Hierarchy(state.data, accessorFn(state.children)).sum(accessorFn(state.size));
```
I guess it will work for my case:
```
 const hierData = d3Hierarchy(state.data, accessorFn(state.children));


PS. Thanks for the fast response!
```

**Comment by vasturiano** [comment]
@Vity01 the `.sum(<fn>)` method is required to compute the sunburst/partition layout, as you can see here:
https://github.com/d3/d3-hierarchy#node_sum

Perhaps the confusing part is in the function name, it's called `sum` but its real purpose is to instruct the layout how to extract values/sizes from each node. Internally, those values are then summed since there is a hierarchy of nodes, but the method simply accesses the value in each node, in your case that should be the `value` property I assume.

So basically a node with children can never be smaller than the sum of all its children. These are the constraints of this type of layout. 
Its dimensions are determined by the sum of its children, plus its own value. So, if you want a non-leaf node to have a size *larger* than the sum of its children, you can simply assign it a value that is equal to that additional quantity (excluding the children sum), if that makes sense.

**Comment by DavidC-75** [comment]
Just following-up on this old question, because it's of particular interest for me.
Let me try to rephrase the original question.

Say we have a node A with two children A1 and A2.

By default, the `value` of node A (which determines the size of the arc on the chart) is calculated as `A.size + A1.size + A2.size`. This is what the `sum` function does: it cumulates the size of all nodes in a bottom-up fashion. (in this case the assumption is that `A.size>=0`, `A1.size>=0` and `A2.size>=0`)

However, depending on how the original data is recorded and what it represents, a different behavior is sometimes desirable, where the `value` of node A is directly indicated by `A.size` (in this case the assumption is that `A.size>=A1.size + A2.size`).
Can this behavior be obtained and how ?

PS: the two different behaviors are also explained and illustrated in https://plotly.com/python/sunburst-charts/#branchvalues

**Comment by vasturiano** [comment]
@DavidC-75 thanks for your comment.

I think the most deterministic and less ambiguous way is to interpret the size property of a node as the "self" portion that's not covered by any children. Otherwise you start hitting the edge cases where a node's size might be smaller than the space required for the children.

But, if you're interested in specifying it that way, all you have to do is do the required inverse math of subtracting the children accumulated sizes from the node values (and ensure that it is at least `0`), and use that value in the component. That should be a relatively simple translation to do, even though it requires you to traverse the tree bottom-up to calc the aggregated sizes.

**Comment by vasturiano** [comment]
@DavidC-75 yes, this summing method is baked-in to [d3Hierarchy](https://github.com/d3/d3-hierarchy#node_sum) and is the only available mean to express node sizes across the hierarchical data structure. And the various layouts (f.e. partition which is used by this Sunburst chart) do expect the input data to follow that structure too.

**Comment by DavidC-75** [comment]
Actually, I've found how to get the desired behavior.

When the size property of a node is the "self" portion and excludes the size of all its descendants, the d3 hierarchy is typically built as:

```
  const root = d3.hierarchy(data)
      .sum(d => d.size)
```

Alternatively, when the size property of a node is the "total" and already includes the size of all its descendants, the d3 hierarchy can simply be built instead as:

```
  const root = d3.hierarchy(data)
      .each(d => d.value = d.data.size)
```
